stix_shifter_modules/guardium/stix_transmission/api_client.py

```diff
@@ -1,250 +1,73 @@
-        proxy = connection.get('proxy')
-        if proxy is not None:
-            proxy_url = proxy.get('url')
-            proxy_auth = proxy.get('auth')
-            if (proxy_url is not None and proxy_auth is not None):
-                self.headers['proxy'] = proxy_url
-                self.headers['proxy-authorization'] = 'Basic ' + proxy_auth
-            if proxy.get('x_forward_proxy', None) is not None:
-                self.headers['x-forward-url'] = 'https://' + \
-                    host_port + '/'  # + endpoint, is set by 'add_endpoint_to_url_header'
-                host_port = proxy.get('x_forward_proxy')
-                if proxy.get('x_forward_proxy_auth', None) is not None:
-                    self.headers['x-forward-auth'] = proxy.get(
-                        'x_forward_proxy_auth')
-                self.headers['user-agent'] = 'UDS'
-                url_modifier_function = self.add_endpoint_to_url_header
-
-        self.client = RestApiClient(host_port,
-                                    None,
-                                    self.headers,
-                                    url_modifier_function,
-                                    cert_verify=connection.get(
-                                        'selfSignedCert', True),
-                                    sni=connection.get('sni', None)
-                                    )
-        self.search_timeout = connection['options'].get('timeout')
-
-    def add_endpoint_to_url_header(self, url, endpoint, headers):
-        # this function is called from 'call_api' with proxy forwarding,
-        # it concatenates the endpoint to the header containing the url.
-        headers['x-forward-url'] += endpoint
-        # url is returned since it points to the proxy for initial call
-        return url
-
-    def add_header_elements(self, key, value):
-        self.headers[key] = value
-        return
-
-    def ping_box(self):
-        # Subroto -- Guardium does not have ping facility
-        # We test if we can get the access token if we can then success = true
-        #
+         # Uncomment when implementing data source API client.
+        # auth = configuration.get('auth')
+        # headers = dict()
+        # headers['X-Auth-Token'] = auth.get('token')
+        # self.client = RestApiClient(connection.get('host'),
+        #                             connection.get('port'),
+        #                             connection.get('cert', None),
+        #                             headers,
+        #                             cert_verify=connection.get('cert_verify', 'True')
+        #                             )
+
+        # Placeholder client to allow dummy transmission calls.
+        # Remove when implementing data source API client.
+        url = "https://"+connection["host"]+":"+connection["port"]
+        self.client = GuardApiClient(connection["options"]["client_id"], url, connection["options"]["client_secret"], \
```

Comment:
I know previously `client_id` and `client_secret` were inside `options` but we can put that back in the top level inside connection object. I already made a comment about that inside `config.json` file. If you make that change you than you should be able to read the property like this-

`connection["client_id"]`
`connection["client_secret"]`

Reply:
U right, change was done.